### PR TITLE
Pipeline: support fine grained aggregate

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:

### What is changed and how it works?

wait for https://github.com/pingcap/tiflash/pull/6934

- replace `builder.appendTransformOp(std::make_unique<ExpressionTransformOp>(exec_status, log->identifier(), before_sort_actions));` by `InterpreterUtils::executeExpression`.
- Add `LocalAggregateTransform` to support fine grained agg.
- Add a fine grained shuffle unit test for agg.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp [hunk 1]
```diff
@@ -48,18 +43,21 @@ void PhysicalAggregationBuild::buildPipelineExecGroup(
         context.getSettingsRef().max_spilled_rows_per_file,
         context.getSettingsRef().max_spilled_bytes_per_file,
         context.getFileProvider());
-
     auto params = AggregationInterpreterHelper::buildParams(
         context,
         before_agg_header,
         concurrency,
-        concurrency,
+        1,
```
Comment (SeaRise) [inline dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp:50]: fix, for non fine grained shuffle, the actual value is 1.
https://github.com/pingcap/tiflash/blob/99e4b2050396fdc41ef5fde4e8e0c9bc12b37472/dbms/src/Flash/Planner/Plans/PhysicalAggregation.cpp#L110

**Review by windtalker — APPROVED** [review]
LGTM

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**